lib/ui/recommendation.js

```diff
@@ -4,6 +4,60 @@
 
 const EXPORTED_SYMBOLS = ['Recommendation'];
 
+
+function RecommendationRow(win, data) {
+  this.win = win;
+  this.data = data;
+}
+
+
+RecommendationRow.prototype = {
+  render: function() {
+    let row = this.win.document.createElement('box');
+
+    let image = this.renderImage();
+    if (!!image) {
+      row.appendChild(image);
+    }
+
+    row.appendChild(this.rendertitle());
+    row.appendChild(this.renderUrl());
+    row.appendChild(this.renderLabel());
+    return row;
+  },
+
+  renderImage: function() {
+    if (this.data.enhancements.hasOwnProperty('domain')) {
+      let image = this.win.document.createElement('image');
+      image.setAttribute('src', this.data.enhancements.domain.logo);
+      return image;
+    }
+    return null;
+  },
+
+  rendertitle: function() {
+    let title = this.win.document.createElement('hbox');
```

Comment:
I'm guessing there are some other XUL element types that we could use, rather than hbox everywhere. This is fine for now, though; maybe file a followup bug for after the 3/15 milestone?